BasalDose/BasalDose.c

```diff
@@ -27,13 +30,25 @@ void BasalDose_DoseTimingInitiate(void)
 	LPC_TIM0->MCR |= 1 << 1; // Reset timer on Match 0.
 	LPC_TIM0->TCR |= 1 << 1; // Manually Reset Timer0 (forced)
 	LPC_TIM0->TCR &=~(1 << 1); // Stop resetting the timer.
-	NVIC_EnableIRQ(TIMER0_IRQn); // Enable Timer0 IRQ
```

Comment:
BasalDose_DoseTimingEnable() first, then reset it. If we don't start it before reset it does nothing.